piet/src/text.rs

```diff
@@ -41,10 +62,29 @@ pub trait TextLayoutBuilder {
 /// - If the text position is not at a code point or grapheme boundary, undesirable behavior may
 /// occur.
 ///
+// TODO update hit testing for line breaking
 pub trait TextLayout {
     /// Measure the advance width of the text.
     fn width(&self) -> f64;
 
+
+    /// Used for changing the width of a text layout. Given a width, returns a [`TextLayout`]
+    /// struct with recalculated lines and line metrics.
+    // TODO: Should this take &self or self? I'd say maybe; perhaps the old struct should be kept around
+    // to help with caching.
+    fn update_width(&self, new_width: f64) -> Self;
+
+    /// Given a line number, return a reference to that line's underlying string.
+    // TODO can I return &str without viral lifetime?
+    fn line_text(line_number: usize) -> String;
```

Comment:
assuming this has a `&self` parameter, I _think_ you should be able to do `fn line_text(&self, line_number: usize) -> &str`, without needing a lifetime in the trait itself.

In addition we _might_ want this to return a `Cow<str>`, in case we want to allow more flexible text storage at some point in the future; I'm thinking specifically if we want to support a rope. But I think we can break the API later, and &str is fine for now.

Reply:
Whoops, I knew returning a reference should have worked, not sure why I forgot `&self` :)

I'll keep the `Cow<str>` in mind.